src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -132,6 +132,27 @@ class CoreWorkerDirectActorTaskSubmitter
   /// Check timeout tasks that are waiting for Death info.
   void CheckTimeoutTasks();
 
+  /// Block the actor_task_io_thread_ until the submitter is shutdown.
+  void WaitForShutdown();
+
+  void Shutdown();
+
+  /// Push a task to corresponding client queue by the actor id.
+  ///
+  /// \param[in] task The task spec to push.
+  /// \param[in] callback  The callback function to be called after the task spec be sent
+  /// out. Only used in test code.
+  /// \return void.
+  void AppendTask(const TaskSpecification &task_spec,
```

Comment:
what's the difference between AppendTask and SubmitTask? should kill one of them?

Reply:
`AppendTask` seems can be removed, i will try.